app/models/test_passage.rb

```diff
@@ -0,0 +1,62 @@
+class TestPassage < ApplicationRecord
+
+  SUCCESS_PERCENT = 85
+
+  belongs_to :user
+  belongs_to :test
+  belongs_to :current_question, class_name: 'Question', optional: true
+
+  before_validation :before_validation_set_first_question, on: :create
+  before_validation :before_validation_set_next_question, on: :update
+
+  def completed?
+    current_question.nil?
+  end
+
+  def accept!(answer_ids)
+    if correct_answer?(answer_ids)
+      self.correct_questions += 1
+    end
+
+
+    save!
+  end
+
+  def current_question_number
+    test.questions.order(:id).where('id <= ?', current_question.id).count
```

Comment:
Very strange thing and I do not understand what for it is using. This can be a big problem if you will have more then 1 test, and why you calculate count of questions depends on id
I think if you want to find next unanswered question for this test, you can search by this question has answer or not